packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -554,6 +554,19 @@ add (const Scalar& alpha,
   using Teuchos::rcp;
   using Teuchos::ParameterList;
   typedef CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>  crs_matrix_type;
+  if(!params.is_null())
+  {
+    if(params->isParameter("Call fillComplete") && !params->get<bool>("Call fillComplete"))
+    {
+      if(A.getRowMap()->getComm()->getRank() == 0)
+      {
+        std::cout << "WARNING: Tpetra::MatrixMatrix::add(): 'Call fillComplete' was explicitly set to false,\n";
```

Comment:
I'm not a fan of printing these messages, since this use case is likely driven by MueLu, not by applications (whose stdout will now be filled with warning messages that they will report as a bug to Trilinos).  The `add` function should do the right thing, whatever that means.  Printing that the behavior changed won't help users.

Reply:
Just to clarify -- if changing the behavior doesn't break anything, I'm OK with changing the behavior.  The issue I have is that applications normally would have no control over the behavior of sparse matrix-matrix multiply.  If you don't intend to support this case, I would rather that you threw an exception with an informative message.

Reply:
@mhoemmen OK, I will throw an exception then. Nothing in Trilinos sets "Call fillComplete" = false and then calls this version (which they don't and shouldn't). But for the apps, just like we have warnings for explicitly-enabled things in CMake getting disabled automatically, I don't want this to just continue silently.

Reply:
@brian-kelley Thanks Brian -- I don't mean to be a blocker; I just want to make sure that we can quickly find any issues with the semantics change.  I doubt there will be, but it's likely that we don't have good downstream testing of all the possibilities.